core/common/src/annotation/TextField.ts

```diff
@@ -0,0 +1,179 @@
+  json?: {
+    /** Property names and/or array indices describing the path to the ultimate JSON property. */
+    accessors: Array<string | number>;
+    /** Optionally specifies the expected type of the JSON property. If unspecified, then numbers are treated as "quantity", booleans as "boolean", and strings as "string".
+     * @note This is permitted to be any string because we anticipate adding new [[FieldPropertyType]]s in the future. If you specify an unrecognized type, the
+     * field will fail to format and will display as [[FieldRun.invalidContentIndicator]].
+     */
+    type?: FieldPropertyType | string; // eslint-disable-line @typescript-eslint/no-redundant-type-constituents
+  };
+}
+
+/** Describes the source of the property value against which a [[FieldPropertyPath]] is evaluated.
+ * A field property is always hosted by an [Element]($backend). It may be a property of the element's BIS class itself,
+ * or that of one of its [ElementAspect]($backend)s.
+ * The [[schemaName]] and [[className]] should always identify the exact class that contains [[FieldPropertyPath.propertyName]] - not a subclass thereof.
+ * @beta
+ */
+export interface FieldPropertyHost {
+  /** The Id of the [Element]($backend) that hosts the property. */
+  elementId: Id64String;
+  /** The name of the schema containing the class identified by [[className]]. */
+  schemaName: string;
+  /** The name of the exact class (not a subclass) containing the property identified by [[FieldPropertyPath.propertyName]]. */
+  className: string;
+}
+
+/** As part of [[CoordinateFieldFormatOptions]], specifies which component(s) to include in the display string.
+ * @beta
+ */
+export type CoordinateComponentSelector = "X" | "Y" | "Z" | "XY" | "XYZ";
+
+/** As part of [[FieldFormatOptions]], specifies how to modify the case of the display string.
+ * "as-is" leaves it unmodified. "upper" and "lower" convert it to all upper-case or all lower-case, respectively.
+ * @beta
+ */
+export type FieldCase = "as-is" | "upper" | "lower";
+
+/** As part of a [[FieldFormatOptions]], specifies how to format properties of [[FieldPropertyType]] "boolean".
+ * @beta
+ */
+export interface BooleanFieldFormatOptions {
+  /** The display string if the property is `true`.
+   * Default: "1".
+   */
+  trueString?: string;
+  /** The display string if the property is `false`.
+   * Default: "0".
+   */
+  falseString?: string;
+}
+
+/** As part of a [[FieldFormatOptions]], specifies how to format properties of [[FieldPropertyType]] "coordinate".
+ * @beta
+ */
+export interface CoordinateFieldFormatOptions {
+  /** Specifies which component(s) (X, Y, and/or Z) to include in the display string.
+   * Default: "XYZ".
+   * @note For two-dimensional points, Z is always omitted even if specified here.
+   */
+  components?: CoordinateComponentSelector;
+  /** The string to insert between components.
+   * Default: "," (comma).
+   */
+  componentSeparator?: string;
+}
```

Comment:
Do you expect that any shared default is defined in project props somewhere or that they will be entered for each field if customized?

Reply:
@Josh-Schifter  or @Alfonso-Martello  where do you plan to grab this kind of formatting information from?

Reply:
What if anything do you want changed here @ColinKerr?